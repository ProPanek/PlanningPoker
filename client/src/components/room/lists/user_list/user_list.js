import React, {Component} from "react"
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";
import Button from "@material-ui/core/Button"
import {decorate, observable} from "mobx";
import DeleteIcon from '@material-ui/icons/Delete'
import {TextField} from "@material-ui/core";


const DefaultSelect = styled.select`
  width:100%;
`

const RoomName = styled.div`
  && {
  margin: 0 auto;
  }
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-wrap: wrap;
`;

const UserDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;

const StyledButton = styled(Button)`
  &&{
  width:100%;
  }
`;
const StyledTextField = styled(TextField)`
  &&{
  width:100%;
  }
`;


class UserList extends Component {
  handleSelect = e => {
    this.userId = e.target.value
  };

  handleKick = () => {
    if (this.userId) {
      this.props.store.kickUser(this.userId)
    }
  }
  handleAdmin = () => {
    if (this.userId) {
      this.props.store.changeAdmin(this.userId)
    }
  }

  handleDelete = () => {
    if (this.roomPassword !== "" && this.props.store.room.roomId !== "") {
      this.props.store.deleteRoom(
        this.props.store.room.roomId,
        this.roomPassword
      );
    }
  };

  handleChange = e => {
    if (e.target.id === "room-password") {
     this.roomPassword = e.target.value
    }
  };

  render() {
    return (
      <UserDiv>
        <RoomName variant="body1" color="inherit">
          {this.props.store.user.userName !== "" && <div> User Name: {this.props.store.user.userName}</div>}
          {this.props.store.room.roomName !== "" && <div> Room Name: {this.props.store.room.roomName}</div>}
        </RoomName>
        <Typography>users : {this.props.store.user.users.length}</Typography>
        <DefaultSelect size={this.props.store.user.users.length} onClick={this.handleSelect}>
          {this.props.store.user.users.length > 0 &&
          this.props.store.user.users.map((data, index) => {
            return (
              <option key={index} value={data.userId}>
                {data.userName}
              </option>
            );
          })}
        </DefaultSelect>
        {this.props.store.user.admin && (
            <Wrapper>
              <StyledButton onClick={this.handleKick}>Kick User</StyledButton>
              <StyledButton onClick={this.handleAdmin}>Give admin </StyledButton>
              <StyledTextField
                id="room-password"
                label="Room Password"
                value={this.roomPassword}
                onChange={this.handleChange}
                type="password"
                margin="normal"
              />
              <StyledButton color="secondary" variant="contained" onClick={this.handleDelete}>
                Delete room
                <DeleteIcon style={{marginLeft: "10px"}}/>
              </StyledButton>
            </Wrapper>

        )}
      </UserDiv>
    )
  }
}

decorate(UserList, {
  userId: observable,
  roomPassword: observable
});

export {UserList}
export default inject("store")(withRouter(observer(UserList)));

