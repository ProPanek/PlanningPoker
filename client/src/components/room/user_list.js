import React, {Component} from 'react'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";
import Button from "@material-ui/core/Button"
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";


const StyledSelect = styled(Select)`
  width:100%;
`

const DefaultSelect = styled.select`
  width:100%;
`

const RoomName = styled.div`
  && {
  margin: 0 auto;
  }
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-wrap: wrap;
`;

const UserDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;
const JiraDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;

class UserList extends Component {
  state = {
    userId: "",
    selectBoardId: "",
    board: "",
    issue: ""
  }

  handleSelect = e => {
    this.setState({userId: e.target.value});
  };

  handleKick = () => {
    if (this.state.userId !== "") {
      this.props.store.kickUser(this.state.userId)
    }
  }
  handleAdmin = () => {
    if (this.state.userId !== "") {
      this.props.store.changeAdmin(this.state.userId)
    }
  }

  selectBoard = (e) => {
    this.props.store.jira.boardId = e.target.value
    this.props.store.selectBoard(e.target.value)
  }

  selectIssue = (e) => {
    const fields = e.target.value.split("_")
    this.props.store.jira.title = fields[0]
    this.props.store.jira.description = fields[1]
    this.props.store.jira.issueId = fields[2]
    this.props.store.broadcastTitle()
    this.props.store.broadcastDescription()
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value});
  };

  render() {
    return (
      <React.Fragment>
        <UserDiv>
          <RoomName variant="body1" color="inherit">
            {this.props.store.user.userName !== "" && <div> User Name: {this.props.store.user.userName}</div>}
            {this.props.store.room.roomName !== "" && <div> Room Name: {this.props.store.room.roomName}</div>}
          </RoomName>
          <Typography>users : {this.props.store.user.users.length}</Typography>
          <DefaultSelect size={this.props.store.user.users.length} onChange={this.handleSelect}>
            {this.props.store.user.users.length > 0 &&
            this.props.store.user.users.map((data, index) => {
              return (
                <option key={index} value={data.userId}>
                  {console.log(data)}
                  {data.userName}
                </option>
              );
            })}
          </DefaultSelect>
          {this.props.store.user.admin && (
            <Wrapper>
              <Button onClick={this.handleKick}>Kick User</Button>
              <Button onClick={this.handleAdmin}>Give admin </Button>
            </Wrapper>
          )}
        </UserDiv>
        <JiraDiv>
          <Typography>Jira Task Picker</Typography>
          {this.props.store.jira.jiraBoards.values.length > 0 &&
          <StyledSelect
            inputProps={{
              name: 'board',
              id: 'board'
            }}
            value={this.state.board} onChange={(e) => {
            this.selectBoard(e);
            this.handleChange(e);
          }}>

            {this.props.store.jira.jiraBoards.values.map((data, index) => {
              return (
                <MenuItem key={index} value={data.id}>
                  {data.name}
                </MenuItem>
              );
            })}
          </StyledSelect>}
          {this.props.store.jira.activeBoard.issues.length > 0 &&
          <StyledSelect
            inputProps={{
              name: 'issue',
              id: 'issue'
            }}
            value={this.state.issue}
            onChange={(e) => {
              this.selectIssue(e);
              this.handleChange(e);
            }}>
            {this.props.store.jira.activeBoard.issues.map((data, index) => {
              return (
                <MenuItem key={index} value={`${data.fields.summary}_${data.fields.description}_${data.id}`}>
                  {data.key} - {data.fields.summary}
                  {console.log(data)}
                </MenuItem>
              );
            })}
          </StyledSelect>}
        </JiraDiv>
      </React.Fragment>
    )
  }
}


export default inject("store")(withRouter(observer(UserList)));

