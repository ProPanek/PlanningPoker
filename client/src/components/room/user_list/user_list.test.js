import React from "react"
import {UserList} from "./user_list"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("UserList", () => {
  it("renders without errors", () => {
    const store = {
      room: { cardsAreTheSame : false, cardResults : [], cardHistory: []},
      user: {admin: true, users: []},
      jira: {activeBoardFetching: false, activeBoard: { issues: []}}
    }

    const component = shallow(<UserList store={store} />)
    expect(component).toMatchSnapshot()
  })
})
