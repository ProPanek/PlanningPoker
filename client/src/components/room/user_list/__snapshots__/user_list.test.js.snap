// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserList
    store={
      Object {
        "jira": Object {
          "activeBoard": Object {
            "issues": Array [],
          },
          "activeBoardFetching": false,
        },
        "room": Object {
          "cardHistory": Array [],
          "cardResults": Array [],
          "cardsAreTheSame": false,
        },
        "user": Object {
          "admin": true,
          "users": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <ForwardRef
            color="inherit"
            variant="body1"
          >
            <div>
               User Name: 
            </div>
            <div>
               Room Name: 
            </div>
          </ForwardRef>
          <WithStyles(Typography)>
            users : 
            0
          </WithStyles(Typography)>
          <ForwardRef
            onClick={[Function]}
            size={0}
          />
          <ForwardRef>
            <WithStyles(Button)
              onClick={[Function]}
            >
              Kick User
            </WithStyles(Button)>
            <WithStyles(Button)
              onClick={[Function]}
            >
              Give admin 
            </WithStyles(Button)>
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              color="inherit"
              variant="body1"
            >
              <div>
                 User Name: 
              </div>
              <div>
                 Room Name: 
              </div>
            </ForwardRef>,
            <WithStyles(Typography)>
              users : 
              0
            </WithStyles(Typography)>,
            <ForwardRef
              onClick={[Function]}
              size={0}
            />,
            <ForwardRef>
              <WithStyles(Button)
                onClick={[Function]}
              >
                Kick User
              </WithStyles(Button)>
              <WithStyles(Button)
                onClick={[Function]}
              >
                Give admin 
              </WithStyles(Button)>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div>
                   User Name: 
                </div>,
                <div>
                   Room Name: 
                </div>,
              ],
              "color": "inherit",
              "variant": "body1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " User Name: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " User Name: ",
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " Room Name: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " Room Name: ",
                  undefined,
                ],
                "type": "div",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
  && {
  margin: 0 auto;
  }
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "users : ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "users : ",
              0,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": false,
              "onClick": [Function],
              "size": 0,
            },
            "ref": null,
            "rendered": false,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "rules": Array [
                  "
  width:100%;
",
                ],
              },
              "displayName": "styled.select",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "select",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  onClick={[Function]}
                >
                  Kick User
                </WithStyles(Button)>,
                <WithStyles(Button)
                  onClick={[Function]}
                >
                  Give admin 
                </WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Kick User",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Kick User",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Give admin ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Give admin ",
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-wrap: wrap;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  height:50%;
  overflow-y:auto;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            false,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  height:50%;
  overflow-y:auto;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <ForwardRef
              color="inherit"
              variant="body1"
            >
              <div>
                 User Name: 
              </div>
              <div>
                 Room Name: 
              </div>
            </ForwardRef>
            <WithStyles(Typography)>
              users : 
              0
            </WithStyles(Typography)>
            <ForwardRef
              onClick={[Function]}
              size={0}
            />
            <ForwardRef>
              <WithStyles(Button)
                onClick={[Function]}
              >
                Kick User
              </WithStyles(Button)>
              <WithStyles(Button)
                onClick={[Function]}
              >
                Give admin 
              </WithStyles(Button)>
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                color="inherit"
                variant="body1"
              >
                <div>
                   User Name: 
                </div>
                <div>
                   Room Name: 
                </div>
              </ForwardRef>,
              <WithStyles(Typography)>
                users : 
                0
              </WithStyles(Typography)>,
              <ForwardRef
                onClick={[Function]}
                size={0}
              />,
              <ForwardRef>
                <WithStyles(Button)
                  onClick={[Function]}
                >
                  Kick User
                </WithStyles(Button)>
                <WithStyles(Button)
                  onClick={[Function]}
                >
                  Give admin 
                </WithStyles(Button)>
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <div>
                     User Name: 
                  </div>,
                  <div>
                     Room Name: 
                  </div>,
                ],
                "color": "inherit",
                "variant": "body1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " User Name: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " User Name: ",
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " Room Name: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " Room Name: ",
                    undefined,
                  ],
                  "type": "div",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
  && {
  margin: 0 auto;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "users : ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                "users : ",
                0,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": false,
                "onClick": [Function],
                "size": 0,
              },
              "ref": null,
              "rendered": false,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "rules": Array [
                    "
  width:100%;
",
                  ],
                },
                "displayName": "styled.select",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "select",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <WithStyles(Button)
                    onClick={[Function]}
                  >
                    Kick User
                  </WithStyles(Button)>,
                  <WithStyles(Button)
                    onClick={[Function]}
                  >
                    Give admin 
                  </WithStyles(Button)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Kick User",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Kick User",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Give admin ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Give admin ",
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-wrap: wrap;
",
                  ],
                },
                "displayName": "styled.div",
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  height:50%;
  overflow-y:auto;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              false,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  height:50%;
  overflow-y:auto;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
